contentctl/actions/inspect.py

```diff
@@ -1,78 +1,88 @@
 import sys
-
-
 from dataclasses import dataclass
-
 import pathlib
 import json
 import datetime
+import timeit
+import time
 
-
-from contentctl.objects.config import inspect
 from requests import Session, post, get
 from requests.auth import HTTPBasicAuth
-import timeit
-import time
+
+from contentctl.objects.config import inspect
+from contentctl.objects.savedsearches_conf import SavedsearchesConf
+from contentctl.objects.errors import (
+    MetadataValidationError,
+    DetectionIDError,
+    DetectionMissingError,
+    VersionDecrementedError,
+    VersionBumpingError
+)
+
+
 @dataclass(frozen=True)
 class InspectInputDto:
-    config:inspect
+    config: inspect
 
 
+# TODO (cmcginley): how should detection version checking behave when it's run on just public, or
+#   just private content?
+# TODO (cmcginley): add `downloads/` to security_content .gitignore
 class Inspect:
```

Comment:
Do we expect `inspect` to ever be run on JUST public or JUST private content? Or, do we expect any external users to use this feature on something other than ESCU? If so, do we need to provide a flag to suppress metadata checking and version enforcement?

Reply:
Added a flag `--enable-metadata-validation`, which defaults to `False`

Without flag:
```bash
contentctl inspect --splunk-api-password SCRUBBED --splunk-api-username SCRUBBED --stack-type classic --enrichments --verbose 
...
AppInspect found [error,failure,manual_check,warning] that MAY cause a failure during AppInspect API:
 - 9 warnings
Details:
 - warning [check_app_configuration_file: check_custom_conf_replication]
 - warning [check_app_configuration_file: check_for_updates_disabled]
 - warning [check_app_configuration_file: check_for_valid_package_id]
 - warning [check_bias_language: check_for_bias_language]
 - warning [check_malware: check_hostnames_and_ips]
 - warning [check_meta_files: check_kos_are_accessible]
 - warning [check_packaging_standards: check_valid_version_number]
 - warning [check_packaging_standards: check_version_is_valid_semver]
 - warning [check_slim_ssai: check_custom_confs]
🟡 Detection metadata validation disabled, skipping.
```

With flag:
```bash
contentctl inspect --splunk-api-password SCRUBBED --splunk-api-username SCRUBBED --stack-type classic --enrichments --verbose 
...
AppInspect found [error,failure,manual_check,warning] that MAY cause a failure during AppInspect API:
 - 9 warnings
Details:
 - warning [check_app_configuration_file: check_custom_conf_replication]
 - warning [check_app_configuration_file: check_for_updates_disabled]
 - warning [check_app_configuration_file: check_for_valid_package_id]
 - warning [check_bias_language: check_for_bias_language]
 - warning [check_malware: check_hostnames_and_ips]
 - warning [check_meta_files: check_kos_are_accessible]
 - warning [check_packaging_standards: check_valid_version_number]
 - warning [check_packaging_standards: check_version_is_valid_semver]
 - warning [check_slim_ssai: check_custom_confs]
Downloading latest ESCU build from Splunkbase to serve as previous build during validation...
Latest release downloaded from Splunkbase to: downloads/escu_latest.tgz

Detection Metadata Validation:
        ✅ Detection metadata looks good and all versions were bumped appropriately :)
```

Reply:
@pyth0n1c please mark as resolved if this looks good :)